pkg/analysis/optionalfields/analyzer.go

```diff
@@ -142,7 +149,22 @@ func (a *analyzer) checkFieldProperties(pass *analysis.Pass, field *ast.Field, f
 	}
 
 	// The pointer preference is now when required.
+	// Validate for omitzero policy.
+	if a.omitZeroPolicy != OptionalFieldsOmitZeroPolicyForbid {
+		if a.omitZeroPolicy != OptionalFieldsOmitZeroPolicyIgnore || hasOmitZero {
+			// If we require omitzero, or the field has omitzero, we can check the field properties based on it being an omitzero field.
+			a.checkFieldPropertiesWithOmitZeroRequired(pass, field, fieldName, jsonTags, hasOmitZero, isPointer, isStruct, hasValidZeroValue)
+		} else {
+			// The field does not have omitzero, and does not require it.
+			a.checkFieldPropertiesWithoutOmitZero(pass, field, fieldName, underlying, jsonTags, hasValidZeroValue, isPointer, isStruct, hasOmitEmpty, completeValidation)
+		}
+	} else {
+		// when the omitzero policy is set to forbid, we need to report removing omitzero if set on the struct fields.
+		a.checkFieldPropertiesWithOmitZeroForbidPolicy(pass, field, fieldName, isStruct, hasOmitZero, jsonTags)
+	}
```

Comment:
A switch might be better here
```suggestion
	switch {
	case a.omitZeroPolicy == OptionalFieldsOmitZeroPolicyForbid:
		// when the omitzero policy is set to forbid, we need to report removing omitzero if set on the struct fields.
		a.checkFieldPropertiesWithOmitZeroForbidPolicy(pass, field, fieldName, isStruct, hasOmitZero, jsonTags)
	case a.omitZeroPolicy == OptionalFieldsOmitZeroPolicyIgnore && !hasOmitZero:
		// The field does not have omitzero, and does not require it.
		a.checkFieldPropertiesWithoutOmitZero(pass, field, fieldName, underlying, jsonTags, hasValidZeroValue, isPointer, isStruct, hasOmitEmpty, completeValidation)
	default:
		// If we require omitzero, or the field has omitzero, we can check the field properties based on it being an omitzero field.
		a.checkFieldPropertiesWithOmitZeroRequired(pass, field, fieldName, jsonTags, hasOmitZero, isPointer, isStruct, hasValidZeroValue)
	}
```